src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -151,4 +161,36 @@ public function getNamespace()
     {
         return 'http://symfony.com/schema/dic/twig';
     }
+
+    private function cleanUpDefinitionsMissingRequiredDependencies(ContainerBuilder $container)
+    {
+        if (!interface_exists('Symfony\Component\Templating\TemplateReferenceInterface')) {
+            $container->removeDefinition('twig.controller.exception');
+        }
```

Comment:
This one looks wrong to me as the Exception controller only depends on Twig, not the Templating component.